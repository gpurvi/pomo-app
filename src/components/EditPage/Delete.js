import React from 'react';
import {withModal} from "../common/Modal";
import {deleteSessions} from "../common/apiCalls";

export class Delete extends React.Component {
    constructor(props) {
        super(props);

        this.onClickHandler = this.onClickHandler.bind(this);
    }

    async onClickHandler(e) {
        e.preventDefault();
        try {
            await deleteSessions(this.props.id);
            this.props.getSessions();
        } catch (err) {
            this.props.onError(err.message);
        }
        this.props.closeModal();
    }

    render() {
        return (
            <div>
                <h2>Delete</h2>
                <p>Delete session {this.props.initValue}</p>
                <button onClick={this.onClickHandler}>Delete</button>
                <button onClick={this.props.closeModal}>Cancel</button>
            </div>
        );
    }
}

export default withModal(Delete);