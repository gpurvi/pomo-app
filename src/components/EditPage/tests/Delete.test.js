import React from 'react';
import {shallow} from 'enzyme';
import {Delete} from '../Delete';

jest.mock('./../../common/apiCalls');

it('renders Delete', () => {
    const wrapper = shallow(<Delete  initValue=''/>);
    expect(wrapper).toMatchSnapshot();
});

describe('onClickHandler',  () => {
    it('calls getSessions', async() => {
        const getSessions = jest.fn();
        const closeModal = jest.fn();
        const wrapper = shallow(<Delete
            closeModal={closeModal}
            getSessions={getSessions}
            id={1}/>);
        await wrapper.find('button').first().simulate('click', {
            preventDefault() {
            }
        });
        expect(getSessions).toHaveBeenCalled();
        expect(closeModal).toHaveBeenCalled();
    });

    // when impelemnt error hangling then impelemnt tests
    // it('calls getSessions', async() => {
    //     const getSessions = jest.fn();
    //     const closeModal = jest.fn();
    //     const wrapper = shallow(<Delete
    //         closeModal={closeModal}
    //         getSessions={getSessions} id={1}/>);
    //     await wrapper.find('button').first().simulate('click', {
    //         preventDefault() {
    //         }
    //     });
    //     // await wrapper.instance().onClickHandler({preventDefault(){}});
    //     expect(getSessions).toHaveBeenCalled();
    // });
});