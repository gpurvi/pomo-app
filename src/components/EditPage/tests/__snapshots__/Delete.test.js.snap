// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Delete 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Delete
    initValue=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Delete
</h2>,
        <p>
          Delete session 
          
</p>,
        <button
          onClick={[Function]}
>
          Delete
</button>,
        <button
          onClick={undefined}
>
          Cancel
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
        },
        "ref": null,
        "rendered": "Delete",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Delete session ",
            "",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Delete session ",
          "",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Delete",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Delete",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cancel",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Cancel",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Delete
</h2>,
          <p>
            Delete session 
            
</p>,
          <button
            onClick={[Function]}
>
            Delete
</button>,
          <button
            onClick={undefined}
>
            Cancel
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
          },
          "ref": null,
          "rendered": "Delete",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Delete session ",
              "",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Delete session ",
            "",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Delete",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Delete",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cancel",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Cancel",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
